src/yunohost/app.py

```diff
@@ -909,10 +909,6 @@ def app_checkurl(auth, url, app=None):
         raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
 
     if domain in apps_map:
-        # Domain already has apps on sub path
-        if path == '/':
-            raise MoulinetteError(errno.EPERM,
-                                  m18n.n('app_location_install_failed'))
```

Comment:
When the restore script is running the app settings are already set up inside /etc/yunohost/apps/APPID/settings.yml . So `domain in apps_map` is always true, because the app is "installed" on this domain...
When we use `yunohost checkurl unstable.local/ -a APPID` we want to check no **other** app than APPID are installed on the root path or a subpath.
But with this line we check all app even APPID itself !